src/main/java/org/opentripplanner/routing/algorithm/filterchain/ItineraryFilterChainBuilder.java

```diff
@@ -171,19 +171,6 @@ public ItineraryFilter build() {
             filters.add(new TransitGeneralizedCostFilter(transitGeneralizedCostLimit));
         }
 
-        // Remove itineraries if max limit is set
-        if (maxNumberOfItineraries > 0) {
-            // Sort first to make sure we keep the most relevant itineraries
-            filters.add(new OtpDefaultSortOrder(arriveBy));
-            filters.add(
-                new MaxLimitFilter(
-                    "number-of-itineraries-filter",
-                    maxNumberOfItineraries,
-                    maxLimitReachedSubscriber
-                )
-            );
-        }
-
```

Comment:
Keep this block as is, but move it down inform of the final sort. I agree that that is better.